drivers/storage/portworx/portworx.go

```diff
@@ -800,6 +801,36 @@ func setDefaultAutopilotProviders(
 	}
 }
 
+func setDefaultAutopilotSecret(cluster *corev1.StorageCluster) {
```

Comment:
We don't need to add the env to the storagecluster. We should directly add it to the autopilot deployment. This should be in the autopilot component when creating the deployment.

Reply:
i think either way works, but this becomes storagecluster defaults so we can track from storagecluster, it would be confusing if we see storagecluster env is different from deployment env.

Reply:
There are lot of other defaults args, envs, etc that we pass to different components but we don't always reflect them back to the StorageCluster. Ideally StorageCluster should just reflect what the user has added/modified. We should not be populating all our defaults in it if we can avoid it.

Reply:
there are already many properties set to default, looks like they are mixed, let's discuss what should, what should not and why.

Reply:
Yes, unfortunately it is mixed. I think we need a clear definition on what should be exposed to user and what we should not. Ideally we should not populate defaults as much as possible. For the existing fields, there were reasons for some of them to add back to storagecluster. 
At least in this case we should not expose it to the user in the storagecluster because it is internal. When security is enabled, the user should just see the security section and should have to worry about envs in other components.

Reply:
resolved